src/argparser.cpp

```diff
@@ -0,0 +1,124 @@
+//
+// Created by Sebastian Schmitt on 19.03.2020.
+//
+
+#include "argparser.h"
+
+#include <utility>
+#include <iomanip>
+
+namespace argparser {
+    std::string Argument::getDescription() {
+        return this->description;
+    }
+
+    std::string Argument::getName() {
+        return this->name;
+    }
+
+    bool Argument::isFound() {
+        return this->_found;
+    }
+
+    Argument *Argument::setRequired(bool required) {
+        this->_required = required;
+        return this;
+
+    }
+
+    Argument::Argument(std::string name, std::string description) {
+        this->name = std::move(name);
+        this->description = std::move(description);
+    }
+
+    ArgumentParser::ArgumentParser(std::string app_name) {
+        this->app_name = std::move(app_name);
+        this->_arguments.push_back(&this->_help);
+    }
+
+    void ArgumentParser::printHelp() {
+        std::cout << "--- " << app_name << " ---" << std::endl;
+        std::cout << "Allowed arguments: " << std::endl;
+
+        int longest_name = 0;
+
+        for (auto &_argument : _arguments) {
+            if (_argument->name.size() > longest_name)
+                longest_name = _argument->name.size();
+        }
+
+        for (auto &_argument : _arguments) {
+            std::cout << std::right << std::setw(longest_name)
+                      << (_argument->_required ? "*" + _argument->name : _argument->name)
+                      << ": "
+                      << _argument->description
+                      << std::endl;
+        }
+        std::cout << "Arguments with a * are marked as required" << std::endl;
+    }
+
+    void ArgumentParser::addArgument(Argument *argument) {
+        _arguments.push_back(argument);
+    }
+
+    void ArgumentParser::parseArguments(int argc, char *argv[]) {
+        if (argc == 1) return;
+
+
+        std::string all_args;
+        for (int i = 1; i < argc; i++) {
+            all_args += std::string(argv[i]);
+            all_args += " ";
+        }
+
+        std::vector<std::string> all_names;
```

Comment:
In .cpp files you can you can use `using namespace std` below the import statements, in order to reduce the verbosity and improve readability.

Reply:
I know. But I didn't want to use this to make clear which namespace we are using here.

Reply:
Okey, that's fine for me.